src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionUtils.java

```diff
@@ -119,6 +122,49 @@ public static HostInfo copyWithAdditionalProps(
         mergedProps);
   }
 
+  public static HostInfo createHostWithProperties(HostInfo baseHost, Map<String, String> properties) {
+    Map<String, String> propertiesCopy = new HashMap<>(properties);
+    propertiesCopy.putAll(baseHost.getHostProperties());
+    String hostEndpoint = baseHost.getHost();
+    int port = baseHost.getPort();
+    String user = propertiesCopy.get(PropertyKey.USER.getKeyName());
+    String password = propertiesCopy.get(PropertyKey.PASSWORD.getKeyName());
+    propertiesCopy.remove(PropertyKey.USER.getKeyName());
+    propertiesCopy.remove(PropertyKey.PASSWORD.getKeyName());
+
+    ConnectionUrl hostUrl = ConnectionUrl.getConnectionUrlInstance(
+        getUrlFromEndpoint(
+            hostEndpoint,
+            port),
+        new Properties());
```

Comment:
Why are you using an empty properties here

Reply:
Before in the AuroraTopologyService, host URLs were created with empty properties, so I kept that behaviour here.